src/esp/gfx/Viewer.h

```diff
@@ -61,8 +61,11 @@ class Viewer : public Magnum::Platform::Application {
                                    const Magnum::Vector2i& position);
 
   assets::ResourceManager resourceManager_;
-  std::shared_ptr<physics::PhysicsManager> physicsManager_;
+  // SceneManager must be before physicsManager_ as the physicsManager_
+  // assumes that it "owns" things owned by the scene manager
```

Comment:
Sounds like double ownership. Did you mean expect instances already initiated in `SceneManager`?

Reply:
No.  It's kinda double ownership.  The physics manager's logic is simpler if it assumes it owns the magnum scene nodes it manipulates.  This is a fine thing to assume as long as the scene graph those nodes are in has a longer lifetime than the physics manager.